### PR TITLE
Jepsen job to reuse builds

### PR DESCRIPTION
**Pull request by maxknv**
#no-merge-commit

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

- support for the jepsen job to take build urls from build report
- use latest master clickhouse in the cron jepsen job

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 3e19a170ca190ecc7454842088b59e52594c2637 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58881/3e19a170ca190ecc7454842088b59e52594c2637/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58881/3e19a170ca190ecc7454842088b59e52594c2637/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: .github/workflows/jepsen.yml [hunk 1]
```diff
@@ -7,8 +7,6 @@ concurrency:
 on: # yamllint disable-line rule:truthy
   schedule:
     - cron: '0 */6 * * *'
-  workflow_dispatch:
-  workflow_call:
```
Comment (Felixoid) [inline .github/workflows/jepsen.yml:11]: we could let the `call` go, but `dispatch` is useful to have this button
image

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -966,13 +966,20 @@ jobs:
...
+  # When it's skipped, all dependent tasks will be skipped too.
+  # DO NOT add it there
   Jepsen:
-    # This is special test NOT INCLUDED in FinishCheck
-    # When it's skipped, all dependent tasks will be skipped too.
-    # DO NOT add it there
-    if: ${{ !failure() && !cancelled() && contains(github.event.pull_request.labels.*.name, 'jepsen-test') }}
+    # we need concurrency as the job uses dedicated instances in the cloud
+    concurrency:
+      group: jepsen
```
Comment (Felixoid) [inline .github/workflows/pull_request.yml:975]: Didn't know the concurrency could go on the `job` level, btw

File: tests/ci/pr_info.py [hunk 3]
```diff
@@ -243,8 +244,11 @@ def __init__(
                         )
                     )
         else:
-            print("event.json does not match pull_request or push:")
-            print(json.dumps(github_event, sort_keys=True, indent=4))
+            if "schedule" in github_event:
+                self.schedule = True
+            else:
```
Comment (Felixoid) [inline tests/ci/pr_info.py:249]: this `else` unnecessary, I'd say

**Review by Felixoid — COMMENTED** [review]
Let's test it and then merge it together with #58901.

File: tests/ci/jepsen_check.py [hunk 4]
```diff
@@ -212,27 +214,36 @@ def main():
...
+    if pr_info.schedule:
+        # get latest clcikhouse by the static link for latest master buit - get its version and provide permanent url for this version to the jepsen
+        build_url = f"{S3_URL}/{S3_BUILDS_BUCKET}/master/amd64/clickhouse"
+        download_build_with_progress(build_url, Path(TEMP_PATH) / "clickhouse")
+        git_runner.run(f"chmod +x {TEMP_PATH}/clickhouse")
+        sha = git_runner.run(
+            f"{TEMP_PATH}/clickhouse local -q \"select value from system.build_options where name='GIT_HASH'\""
+        )
+        version_full = git_runner.run(
+            f"{TEMP_PATH}/clickhouse local -q \"select value from system.build_options where name='VERSION_FULL'\""
```
Comment (Felixoid) [inline tests/ci/jepsen_check.py:226]: `SELECT version()` works beter here

Comment (mkaynov) [reply tests/ci/jepsen_check.py:226]: is it critical difference?

Comment (Felixoid) [reply tests/ci/jepsen_check.py:226]: The following code will be simpler `¯\_(ツ)_/¯`

`version = ".".join(version_full.split(".", 3)[:2]`

Comment (mkaynov) [reply tests/ci/jepsen_check.py]: fixed

**Pull request merged**